### ISSUE TITLE
Package Chirpstack components for NixOS

### ISSUE THREAD

**Issue by stv0g** [chirpstack/chirpstack#568]
Hi all,

I've started to package Chirpstack for NixOS.

Most of the independency components are already availabel in nixpkgs-unstable:

https://search.nixos.org/packages?channel=unstable&type=packages&query=chirpstack

Only the main server and WebUI are missing: https://github.com/NixOS/nixpkgs/pull/358582

Some feedback or support in getting this would be appreciated :)

Once, we have everthing packaged, I would like to continue creating a NixOS module as well :)

**Comment by brocaar** [comment]
I wish I could help you with this. While using NixOS myself, I have zero experience with writing package definitions. Hopefully somebody else could step in and help you with your questions.

**Comment by kaaax0815** [comment]
It seems you already did that for the others? Whats stopping you from doing chirpstack itself? It seems like the definition wouldn't be that different

**Comment by stv0g** [comment]
The Main Server is the most complex complex component as it bundles both front and backend.

But nothing is really stopping us. It's just a lack of time and expertise from my side.

I am still getting started with Chirpstack.. So am confident, we will manage it at some point. Just some feedback would be great 👍

**Comment by brocaar** [comment]
That is correct, see https://github.com/chirpstack/chirpstack/blob/master/chirpstack/src/api/mod.rs#L98. This means you need to build the UI **before** compiling the `chirpstack` binary.

**Comment by thatwhichisdev** [comment]
Sup lads, any updates here?

I think the network server is the only missing piece on the `nixpkgs`, and honestly it is very needed :)

Additionally I want to ask if there is any plan to produce modules with configurations per service, so for example you can simply specify and let the module install the required package and configure service according to what you supplied. 
```nix
  services.chirpstack-concentratord = {
    enable = true;

    settings = {
    .....
  }
```

For beginning I think each repository can just define it's own `flake.nix` and `nix` folder with the `module.nix`, those who wish to use these packages can simply reference the flake and have a great time using the chirp stack.
